src/design-system/organisms/MintForm/MintForm.tsx

```diff
@@ -0,0 +1,212 @@
+import { MarketTradeType, Token, TokenType } from '../../../interfaces';
+import { roundToTwo, tokenDivideDown, tokenMultiplyUp } from '../../../utils/math';
+
+const endAdornmentStyles: SxProps<Theme> = { whiteSpace: 'nowrap' };
+const TokenPriceDefault = {
+  yes: 0,
+  no: 0,
+};
+export interface MintFormProps {
+  title: string;
+  /**
+   * Callback to get the form values
+   */
+  handleSubmit: (
+    values: FormikValues,
+    formikHelpers: FormikHelpers<FormikValues>,
+  ) => void | Promise<void>;
+  /**
+   * Initial values to use when initializing the form. Default is 0.
+   */
+  initialValues?: FormikValues;
+  /**
+   * Is wallet connected
+   */
+  connected?: boolean;
+  /**
+   * TokenName to display
+   */
+  tokenName?: string;
+  /**
+   * Token Price
+   */
+  tokenPrice?: {
+    yes: number;
+    no: number;
+  };
+  /**
+   * Market Id
+   */
+  marketId: string;
+  /**
+   * User token values
+   */
+  userTokens?: Token[];
+}
+
+export const MintForm: React.FC<MintFormProps> = ({
+  title = 'mintButton',
+  handleSubmit,
+  initialValues,
+  connected,
+  tokenName,
+  marketId,
+  userTokens,
+  tokenPrice = TokenPriceDefault,
+}) => {
+  const { t } = useTranslation('common');
+  const yesTokenId = React.useMemo(() => getYesTokenId(marketId), [marketId]);
+  const noTokenId = React.useMemo(() => getNoTokenId(marketId), [marketId]);
+  const [expectedPosition, setExpectedPosition] = React.useState<PositionItem[]>([]);
+  const [userAmounts, setUserAmounts] = React.useState({
+    yesToken: 0,
+    noToken: 0,
+  });
+
+  React.useEffect(() => {
+    if (userTokens) {
+      const yesToken = getTokenQuantityById(userTokens, yesTokenId);
+      const noToken = getTokenQuantityById(userTokens, noTokenId);
+      setUserAmounts({
+        noToken,
+        yesToken,
+      });
+    }
+  }, [userTokens, yesTokenId, noTokenId]);
+
+  const validationSchema = Yup.object({
+    amount: Yup.number()
+      .min(0.000001, `${t('minMint')} 0.000001`)
+      .required(t('required')),
+  });
+
+  const handleChange = React.useCallback(
+    (e: any) => {
+      if (Number.parseFloat(e.target.value) > 0) {
+        const newVal = e.target.value;
+        if (!connected) {
+          const newPosition: PositionItem[] = [
+            {
+              label: `${t(TokenType.yes)} ${t('tokens')}`,
+              value: newVal,
+            },
+            {
+              label: `${t(TokenType.no)} ${t('tokens')}`,
+              value: newVal,
+            },
+          ];
+          setExpectedPosition(newPosition);
+        } else {
+          const newTotalValue =
+            tokenPrice.yes * (userAmounts.yesToken + tokenMultiplyUp(newVal)) +
+            tokenPrice.no * (userAmounts.noToken + tokenMultiplyUp(newVal));
+          const newPosition: PositionItem[] = [
+            {
+              label: `${t(TokenType.yes)} ${t('tokens')}`,
+              value: `${roundToTwo(tokenDivideDown(userAmounts.yesToken))} (+${newVal})`,
```

Comment:
im slightly confused by this display,
i believe it should show the updated value of our token holdings not our current 
<img width="394" alt="Screen Shot 2021-08-30 at 13 52 06" src="https://user-images.githubusercontent.com/37285713/131335320-91b8bfdd-d8bf-44d9-a4e3-28d54ede9c48.png">

Reply:
<img width="767" alt="Screen Shot 2021-08-30 at 14 06 30" src="https://user-images.githubusercontent.com/37285713/131336501-87392a08-bc46-489a-ba74-4bd4a77553c3.png">
based on these screens ^

Reply:
It seems it shouldn't be updated value. Am I right @shekhar-shubhendu?